### PR TITLE
Non-completed tests should cause failures

### PR DESCRIPTION
**Pull request by grouma**
- Update engine to consider non-completed tests as a failure run
- Update `compact` and `expanded` reporters to list the non-completed tests
- Add corresponding test
- Rev version numbers

Closes https://github.com/dart-lang/test/issues/1003

### CODE REVIEW

File: pkgs/test/test/runner/engine_test.dart [hunk 1]
```diff
@@ -58,6 +59,19 @@ void main() {
     engine.suiteSink.close();
   });
 
+  test("returns failure if all tests do not complete", () async {
```
Comment (natebosch) [inline pkgs/test/test/runner/engine_test.dart]: phrasing is weird. "if any test does not complete" would make more sense to me.

File: pkgs/test/test/runner/engine_test.dart [hunk 2]
```diff
@@ -58,6 +59,19 @@ void main() {
...
+    var engine = declareEngine(() {
+      for (var i = 0; i < 3; i++) {
+        test("test ${i + 1}", () async {
+          await Future.delayed(Duration(seconds: 5));
```
Comment (natebosch) [inline pkgs/test/test/runner/engine_test.dart]: can we use a `Completer`  that we never `.complete()` to make this more explicit than a delay?

File: pkgs/test/test/runner/engine_test.dart [hunk 3]
```diff
@@ -58,6 +59,19 @@ void main() {
...
+        });
+      }
+    });
+
+    unawaited(expectLater(engine.run(), completion(isFalse)));
```
Comment (natebosch) [inline pkgs/test/test/runner/engine_test.dart]: change `unawaited(expectLater(...))` to `expect(...)`

File: pkgs/test/test/runner/engine_test.dart (hunk 2 above)
Comment (grouma) [reply pkgs/test/test/runner/engine_test.dart]: Discussed offline. These tests are odd so I can't open non-completed tasks. We came up with an alternative that I'll implement though.

**Pull request merged**